rd-cpp/thirdparty/clsocket/src/SimpleSocketSender.h

```diff
@@ -0,0 +1,40 @@
+#ifndef SIMPLESOCKETSENDER_H
+#define SIMPLESOCKETSENDER_H
+
+#include <memory>
+
+#include "SimpleSocket.h"
+
+class CSimpleSocketSender
+{
+    std::shared_ptr<CSimpleSocket> m_socket;
+    CSimpleSocket::CSocketError m_error;
+
+public:
+    explicit CSimpleSocketSender(const std::shared_ptr<CSimpleSocket>& socket) : m_error(CSimpleSocket::SocketSuccess)
+    {
+        if (socket->m_nSocketType == CSimpleSocket::CSocketType::SocketTypeTcp)
+            m_socket = socket;
+        else
+            m_socket = std::make_shared<CSimpleSocket>();
```

Comment:
Not sure I understand this verse. Why does it create a new socket if a passed socket is not a TCP one?

Reply:
Because only TCP socket is supported. Technically new socket is always in disconnected state and every attempt to send to this socket will fail because socket isn't valid. We can throw exception instead in that case as an alternative.

Reply:
I would suggest throwing then.